engine/security/src/test/scala/pl/touk/nussknacker/ui/security/oauth2/JwtTokenAuthenticationSpec.scala

```diff
@@ -0,0 +1,85 @@
+package pl.touk.nussknacker.ui.security.oauth2
+
+import akka.http.javadsl.model.headers.HttpCredentials
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.{Directives, Route}
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import com.typesafe.config.ConfigFactory
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport
+import org.scalatest.{FunSpec, Matchers}
+import pdi.jwt.{JwtAlgorithm, JwtCirce, JwtClaim}
+import pl.touk.nussknacker.ui.security.api.AuthenticationResources
+import pl.touk.nussknacker.ui.security.http.RecordingSttpBackend
+import sttp.client.testing.SttpBackendStub
+import sttp.model.Uri
+
+import java.net.URI
+import java.security.KeyPairGenerator
+import java.time.Clock
+import java.util.Base64
+
+class JwtTokenAuthenticationSpec extends FunSpec with Matchers with ScalatestRouteTest with Directives with FailFastCirceSupport {
+
+  implicit val clock: Clock = Clock.systemUTC()
+
+  private val keyPair = KeyPairGenerator.getInstance("RSA").generateKeyPair()
+
+  private val config = ConfigFactory.parseString(
+    s"""
+        authentication: {
+          method: "OAuth2"
+          usersFile: "classpath:oauth2-users.conf"
+          authorizeUri: "http://ignored"
+          clientSecret: "ignored"
+          clientId: "ignored"
+          profileUri: "http://authorization.server/userinfo"
+          accessTokenUri: "http://authorization.server/token"
+          redirectUri: "http://ignored"
+          jwt: {
+            accessTokenIsJwt: true
+            publicKey: "${Base64.getEncoder.encodeToString(keyPair.getPublic.getEncoded)}"
+            audience: "http://nussknacker"
+          }
+        }
+      """.stripMargin)
+
+  private val userinfoUri = Uri(URI.create(config.getString("authentication.profileUri")))
```

Comment:
Won't be easier to use those values inside config block instead of extracting it from config? Currently after changing of layout of configuration your tests will stop working and u will have to correct those paths.